Zend/zend_interfaces_dimensions.stub.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/** @generate-class-entries */
+
+interface ArrayAccess
+{
+    /** @tentative-return-type */
+    public function offsetExists(mixed $offset): bool;
+
+    /**
+     * Actually this should be return by ref but atm cannot be.
+     * @tentative-return-type
+     */
+    public function offsetGet(mixed $offset): mixed;
+
+    /** @tentative-return-type */
+    public function offsetSet(mixed $offset, mixed $value): void;
+
+    /** @tentative-return-type */
+    public function offsetUnset(mixed $offset): void;
+}
+
+interface DimensionReadable
+{
+    public function offsetGet(mixed $offset): mixed;
+
+    public function offsetExists(mixed $offset): bool;
+}
+
+interface DimensionFetchable extends DimensionReadable
+{
+    public function &offsetFetch(mixed $offset): mixed;
+}
+
+interface DimensionWritable
+{
+    public function offsetSet(mixed $offset, mixed $value): void;
+}
+
+interface DimensionUnsetable
+{
+    public function offsetUnset(mixed $offset): void;
+}
+
+interface Appendable
+{
+    public function append(mixed $value): void;
```

Comment:
Can we please consider `offsetAppend` instead, and have the relevant SPL objects proxy their `append` to `offsetAppend`? As I pointed out to you in Stack Overflow chat, there are real-world projects with `append` methods that don't match this signature nor semantics, which will make upgrading more painful (but not a BC break, just a road block to onboarding the new APIs).

Reply:
Please provide this feedback on the internals mailing list.